src/lib/pipe_lib/strict_pipe.ml

```diff
@@ -75,6 +75,8 @@ module Reader = struct
     reader.has_reader <- true ;
     wrap_reader (Pipe.filter_map reader.reader ~f)
 
+  let clear t = Pipe.clear t.reader
```

Comment:
don't you also need to stop the messages? Or is that handled in a different way now?